### PR TITLE
chore(test-data): create `CartDiscount` and associated draft

### PR DESCRIPTION
**Pull request by jaikamat**
## Summary

This PR creates the [`CartDiscount`](https://docs.commercetools.com/api/projects/cartDiscounts#cartdiscount) model.

## Description

More of the same, pretty standard migration stuff.

### CODE REVIEW

File: models/cart-discount/src/generator.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+    description: fake(() => LocalizedString.random()),
+    value: fake((f) =>
+      f.helpers.arrayElement([
+        CartDiscountValueAbsolute.CartDiscountValueAbsoluteDraft.random(),
+        CartDiscountValueFixed.CartDiscountValueFixedDraft.random(),
+        CartDiscountValueGiftLineItem.CartDiscountValueGiftLineItemDraft.random(),
+        CartDiscountValueRelative.CartDiscountValueRelativeDraft.random(),
+      ])
+    ),
+    cartPredicate: '1=1',
```
Comment (jaikamat) [inline models/cart-discount/src/generator.ts:34]: Again, we can deal with the predicate and `references` in presets 👍 No need to increase complexity right now, imo

File: models/cart-discount/src/cart-discount-draft/generator.ts [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+      f.helpers.arrayElement([
+        CartDiscountValueAbsolute.CartDiscountValueAbsoluteDraft.random(),
+        CartDiscountValueFixed.CartDiscountValueFixedDraft.random(),
+        CartDiscountValueGiftLineItem.CartDiscountValueGiftLineItemDraft.random(),
+        CartDiscountValueRelative.CartDiscountValueRelativeDraft.random(),
+      ])
+    ),
+    cartPredicate: '1=1',
+    target: null,
+    sortOrder: fake((f) => String(Math.random())),
```
Comment (stephsprinkle) [inline models/cart-discount/src/cart-discount-draft/generator.ts:30]: This needs to be between 0 and 1

File: models/cart-discount/src/cart-discount-draft/generator.ts [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    isActive: fake((f) => f.datatype.boolean()),
+    validFrom: fake(getValidFrom),
+    validUntil: fake(getValidUntil),
+    requiresDiscountCode: fake((f) => f.datatype.boolean()),
+    stackingMode: fake((f) =>
+      f.helpers.arrayElement(['Stacking', 'StopAfterThisDiscount'])
```
Comment (stephsprinkle) [inline models/cart-discount/src/cart-discount-draft/generator.ts:36]: This should be a constant ([example](https://github.com/commercetools/test-data/blob/main/models/channel/src/constants.ts)), then you could do something like f.helpers.arrayElement([Object.values(stackingMode)]) and we can later reference it as stackingMode.Stacking in a preset.

File: models/cart-discount/src/cart-discount-draft/transformers.ts [hunk 4]
```diff
@@ -0,0 +1,27 @@
+import { LocalizedString } from '@commercetools-test-data/commons';
+import { Transformer } from '@commercetools-test-data/core';
+import type { TCartDiscountDraft, TCartDiscountDraftGraphql } from '../types';
+
+const transformers = {
+  default: Transformer<TCartDiscountDraft, TCartDiscountDraft>('default', {
+    buildFields: ['value', 'name', 'description'],
```
Comment (stephsprinkle) [inline models/cart-discount/src/cart-discount-draft/transformers.ts:7]: Should these also include `target` and `custom` even though we aren't technically doing anything with them now?

File: models/cart-discount/src/cart-discount-draft/transformers.ts [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+  }),
+  graphql: Transformer<TCartDiscountDraft, TCartDiscountDraftGraphql>(
+    'graphql',
+    {
+      buildFields: ['value', 'name', 'description'],
+      addFields: ({ fields }) => ({
+        nameAllLocales: LocalizedString.toLocalizedField(fields.name),
+        descriptionAllLocales: LocalizedString.toLocalizedField(
+          fields.description
+        ),
```
Comment (stephsprinkle) [inline models/cart-discount/src/cart-discount-draft/transformers.ts:20]: I don't see these fields in the GraphQL docs
image

File: models/cart-discount/src/generator.ts [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+    target: null,
+    sortOrder: fake((f) => String(Math.random())),
+    isActive: fake((f) => f.datatype.boolean()),
+    validFrom: fake(getValidFrom),
+    validUntil: fake(getValidUntil),
+    requiresDiscountCode: fake((f) => f.datatype.boolean()),
+    references: [],
+    stackingMode: fake((f) =>
+      f.helpers.arrayElement(['Stacking', 'StopAfterThisDiscount'])
```
Comment (stephsprinkle) [inline models/cart-discount/src/generator.ts:43]: Can use the same constant here too

File: models/cart-discount/src/generator.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+      f.helpers.arrayElement([
+        CartDiscountValueAbsolute.CartDiscountValueAbsoluteDraft.random(),
+        CartDiscountValueFixed.CartDiscountValueFixedDraft.random(),
+        CartDiscountValueGiftLineItem.CartDiscountValueGiftLineItemDraft.random(),
+        CartDiscountValueRelative.CartDiscountValueRelativeDraft.random(),
+      ])
+    ),
+    cartPredicate: '1=1',
+    target: null,
+    sortOrder: fake((f) => String(Math.random())),
```
Comment (stephsprinkle) [inline models/cart-discount/src/generator.ts:36]: Same 0 to 1

File: models/cart-discount/src/transformers.ts [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+const transformers = {
+  default: Transformer<TCartDiscount, TCartDiscount>('default', {
+    buildFields: [
+      'value',
+      'name',
+      'description',
+      'createdBy',
+      'lastModifiedBy',
+    ],
```
Comment (stephsprinkle) [inline models/cart-discount/src/transformers.ts:13]: Same question here regarding fields we aren't technically using yet but could in the future (`target`, `references`, `custom`)

File: models/cart-discount/src/builder.spec.ts [hunk 9]
```diff
@@ -0,0 +1,131 @@
...
+        createdBy: expect.objectContaining({
+          customerRef: expect.objectContaining({ typeId: 'customer' }),
+          userRef: expect.objectContaining({ typeId: 'user' }),
+          __typename: 'Initiator',
+        }),
+        lastModifiedAt: expect.any(String),
+        lastModifiedBy: expect.objectContaining({
+          customerRef: expect.objectContaining({ typeId: 'customer' }),
+          userRef: expect.objectContaining({ typeId: 'user' }),
+          __typename: 'Initiator',
```
Comment (tylermorrisford) [inline models/cart-discount/src/builder.spec.ts:125]: 💪

File: models/cart-discount/src/cart-discount-draft/generator.ts (hunk 2 above)
Comment (jaikamat) [reply models/cart-discount/src/cart-discount-draft/generator.ts]: Thanks! dc3df4853128bf7da52f0700bf3d6a42273ac13a should do it

File: models/cart-discount/src/cart-discount-draft/generator.ts (hunk 3 above)
Comment (jaikamat) [reply models/cart-discount/src/cart-discount-draft/generator.ts]: Great strategy! Resolved in 0ff3350a6e7d3ff9f23e2f131992bd3cec4efc68

File: models/cart-discount/src/cart-discount-draft/transformers.ts (hunk 5 above)
Comment (jaikamat) [reply models/cart-discount/src/cart-discount-draft/transformers.ts]: Resolved in fa1ff8648a9f7b779e0e292bdd12fe0245538fb1

File: models/cart-discount/src/transformers.ts (hunk 8 above)
Comment (jaikamat) [reply models/cart-discount/src/transformers.ts:13]: Interesting, perhaps we could include these when we do eventually use models here? Right now there's nothing to build.

However, I should add these fields to the tests - just seems like good form to do so: see a919301024a2970cddb17ecd94dff85156e94c7d

Comment (stephsprinkle) [reply models/cart-discount/src/transformers.ts:13]: Works for me

File: models/cart-discount/src/types.ts [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+  __typename: 'CartDiscount';
+  createdBy: TClientLoggingGraphql;
+  lastModifiedBy: TClientLoggingGraphql;
+  nameAllLocales?: TLocalizedStringGraphql | null;
+  descriptionAllLocales?: TLocalizedStringGraphql | null;
+};
+export type TCartDiscountDraftGraphql = TCartDiscountDraft & {
+  __typename: 'CartDiscountDraft';
+  nameAllLocales?: TLocalizedStringGraphql | null;
+  descriptionAllLocales?: TLocalizedStringGraphql | null;
```
Comment (stephsprinkle) [inline models/cart-discount/src/types.ts:21]: Don't need these

Comment (jaikamat) [reply models/cart-discount/src/types.ts]: 🦅 3e25e181fcfbe1e99c7d367b15be8f5e8c840f53

**Review by stephsprinkle — APPROVED** [review]
Lovely!

**Review by tylermorrisford — APPROVED** [review]
LGTM!

**Pull request merged**